### PR TITLE
Adding HA to docker installs of contiv

### PR DESCRIPTION
**Pull request by neelimamukiri**
Please ignore the change in scripts/build.sh. Vagrantfile & makefile changes are for testing a multi-master setup. Tested with 3 master & 2 worker nodes and checked that the etcdcluster remained health for up to 2 master node failures.
Tested legacy-swarm & swarm-mode. K8s HA will not be affected by this.

### CODE REVIEW

File: cluster/Vagrantfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ require 'rubygems'
 require 'fileutils'
 require 'yaml'
 
-DEFAULT_K8S_VERSION = 'v1.6.4'.freeze
+DEFAULT_K8S_VERSION = 'v1.6.5'.freeze
```
Comment (gkvijay) [inline cluster/Vagrantfile]: was this change intentional?

File: install/ansible/uninstall.sh [hunk 2]
```diff
@@ -148,13 +148,8 @@ echo "Uninstalling Contiv"
...
 # Uninstall contiv & API Proxy
-if [ $uninstall_v2plugin == true ]; then
-	echo '- include: uninstall_auth_proxy.yml' >$ansible_path/uninstall_plays.yml
-	echo '- include: uninstall_v2plugin.yml' >>$ansible_path/uninstall_plays.yml
-else
-	echo '- include: uninstall_auth_proxy.yml' >$ansible_path/uninstall_plays.yml
-	echo '- include: uninstall_contiv.yml' >>$ansible_path/uninstall_plays.yml
-fi
+echo '- include: uninstall_auth_proxy.yml' >$ansible_path/uninstall_plays.yml
+echo '- include: uninstall_v2plugin.yml' >>$ansible_path/uninstall_plays.yml
```
Comment (gkvijay) [inline install/ansible/uninstall.sh]: Should this be uninstall_contiv.yml?

File: scripts/vagrantup.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -euo pipefail
+
+num_nodes=${CONTIV_NODES-2}
+num_masters=${CONTIV_MASTERS-1}
```
Comment (gkvijay) [inline scripts/vagrantup.sh]: Vagrantfile seems to be using CONTIV_NODES and CONTIV_MASTERS. Here you are leaving a few nodes

File: cluster/Vagrantfile (hunk 1 above)
Comment (neelimamukiri) [inline cluster/Vagrantfile]: Yes, this is the kubectl version now.

File: scripts/vagrantup.sh (hunk 3 above)
Comment (neelimamukiri) [inline scripts/vagrantup.sh]: This is the syntax for the default values (2 for nodes and 1 for masters).

File: install/ansible/uninstall.sh (hunk 2 above)
Comment (neelimamukiri) [inline install/ansible/uninstall.sh]: Good catch! Yes, updated.

File: cluster/Vagrantfile [hunk 4]
```diff
@@ -19,6 +19,11 @@ ORC_SWARM = 'swarm-mode'.freeze
 ORC_KUBEADM = 'kubeadm'.freeze
 ORC_KUBEADM_IDX = 2
 
+num_nodes = (ENV["CONTIV_NODES"] || 2).to_i
+num_masters = (ENV["CONTIV_MASTERS"] || 1).to_i
+num_workers = num_nodes - num_masters
+
+
 # Global vars
 token = 'd900e1.8a392798f13b33a4'
```
Comment (rhim) [inline cluster/Vagrantfile]: nit: we should probably create this randomly. Maybe file a future issue for this.

File: scripts/vagrantup.sh (hunk 3 above)
Comment (gkvijay) [inline scripts/vagrantup.sh]: Don't you need a ":" for default CONTIV_NODES:-2

Comment (dseevr) [inline scripts/vagrantup.sh]: `:-` and `-` are synonymous inside `${}`
